cdc/kv/client.go

```diff
@@ -390,49 +587,180 @@ func (c *CDCClient) divideAndSendEventFeedToRegions(
 	}
 }
 
-// singleEventFeed makes a EventFeed RPC call.
+// handleError handles error returned by a region. If some new EventFeed connection should be established, the region
+// info will be sent to `regionCh`.
+func (c *CDCClient) handleError(ctx context.Context, errInfo regionErrorInfo, regionCh chan<- singleRegionInfo) error {
+	err := errInfo.err
+	switch eerr := errors.Cause(err).(type) {
+	case *eventError:
+		innerErr := eerr.err
+		if notLeader := innerErr.GetNotLeader(); notLeader != nil {
+			eventFeedErrorCounter.WithLabelValues("NotLeader").Inc()
+			// TODO: Handle the case that notleader.GetLeader() is nil.
+			c.regionCache.UpdateLeader(errInfo.verID, notLeader.GetLeader().GetStoreId(), errInfo.rpcCtx.PeerIdx)
```

Comment:
will this panic if `notLeader.GetLeader()` is nil ?

Reply:
I think it's ok, because TiDB didn't check it either, and `GetStoreId`'s implementation checked whether it's nil and returns 0 in that case. When `UpdateLeader` receives 0 as the `StoreId` it will simply try to send the request to the next peer. I think maybe I need to remove the TODO comment.

Reply:
Oh, no. TiDB uses different backoff time according to whether it's nil. We need to process it later, but it won't panic.